test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -496,6 +496,60 @@ void ConfigureTestServices(IServiceCollection services)
             Assert.Equal(shouldEnrichBeCalled, enrichCalled);
         }
 
+        [Fact]
+        public async Task AspNetCoreInstrumentationRespectSuppressInstrumentation()
+        {
+            try
+            {
+                bool isProcessorOnStartCalled = false;
+                bool isProcessorOnEndCalled = false;
+                var activityProcessor = new Mock<BaseProcessor<Activity>>();
+                activityProcessor.Setup(p => p.OnStart(It.IsAny<Activity>())).Callback(() => isProcessorOnStartCalled = true);
+                activityProcessor.Setup(p => p.OnEnd(It.IsAny<Activity>())).Callback(() => isProcessorOnEndCalled = true);
+
+                bool isPropagatorCalled = false;
+                var propagator = new Mock<TextMapPropagator>();
+                propagator.Setup(m => m.Extract(It.IsAny<PropagationContext>(), It.IsAny<HttpRequest>(), It.IsAny<Func<HttpRequest, string, IEnumerable<string>>>()))
+                    .Returns(() =>
+                    {
+                        isPropagatorCalled = true;
+                        return default(PropagationContext);
+                    });
+
+                // Arrange
+                using (var testFactory = this.factory
+                    .WithWebHostBuilder(builder =>
+                        builder.ConfigureTestServices(services =>
+                        {
+                            Sdk.SetDefaultTextMapPropagator(propagator.Object);
+                            this.openTelemetrySdk = Sdk.CreateTracerProviderBuilder()
+                                .AddAspNetCoreInstrumentation()
+                                .AddProcessor(activityProcessor.Object)
+                                .Build();
+                        })))
+                {
+                    using var scope = SuppressInstrumentationScope.Begin();
+                    using var client = testFactory.CreateClient();
+                    var response = await client.GetAsync("/api/values/2");
+                    response.EnsureSuccessStatusCode(); // Status Code 200-299
```

Comment:
Do we need this line?

Reply:
It's not really needed, it's like being extra cautious to ensure that the request payload is not tampered with in anyway under SuppressInstrumentation.

Reply:
I see, makes sense!